src/silx/gui/dialog/AbstractDataFileDialog.py

```diff
@@ -443,8 +443,13 @@ def saveState(self):
         stream.writeInt32(self.__serialVersion)
         stream.writeQVariant(self.__detailView.header().saveState())
         viewMode = self.viewMode()
-        if qt.BINDING == 'PyQt6':  # No auto conversion to int
+        if qt.BINDING == 'PyQt6':
             viewMode = viewMode.value
+        if qt.BINDING == 'PySide6':  # No auto conversion to int
+            from PySide6 import __version__ as pyside6_version
+            from pkg_resources import parse_version as _parse_version
+            if parse_version(__version__) >= parse_version('6.4'):
+                viewMode = viewMode.value
```

Comment:
Why dont use just use `viewMode.value` always? this should be retrocompatible no?

Or the following probably also work, no?
```int(viewMode)```

I mean there is probably a compatible way to get the int in this way.

Reply:
`int(viewMode)` doesn't work..

Reply:
What about
```
if hasattr(viewMode, "value"):
    viewMode = viewMode.value
```

Reply:
yes, should do the work. But as was saying @t20100 this is a major release and to my knowledge the usage of pyside6 is very limited for now. So it would be better / simpler to force `pyside6 > 6.4` as requirements and fix what needs to be fix only for `pyside6 > 6.4` only. This would simplify the management.

Reply:
The issue with `hasattr` is that you don't know for which binding/version it is needed and so it's gonna stay forever.

I would be in favor of dropping support of previous version of `pyside6`, I doubt it is used a lot yet...

Reply:
Contrary to kson, since python 1 you can introcide some comments in the code with `#` ;-)